docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,295 @@
+* The the specific filter that will return only a specific group when the group annotation from a first-class member is injected and the resulting filter is conjoined with the common filter
+* The attribute on the first-class entities in the LDAP server that holds the groups they are a member of
+* If groups are stored as first-class entries in the LDAP server
+* The (optional) specification for how to transfer attributes from a first-class group entity on the LDAP server to fields in an OpenShift `Group`. This is only present if the `firstClassGroups` value above is `true`.
+
+`LDAPSyncOptions` needs the following schema for LDAP schema where group membership queries will return group entries:
+```yaml
+commonFilter: "OU=OpenShiftDevelopers,OU=Acme,DC=acme,DC=com" (1)
+specificFilter:
+  group: (2)
+    key: "OU=Groups,%s"
+    openShiftGroup: (3)
+      - cn: "name"
+      - ou: "annotations.organzationalUnit"
+    ldapGroup: (4)
+      value: "CN=%s"
+  member: (5)
+    key: "DN=%s,OU=Users"
+memberAttribute: "member" (6)
+groupFields: (7)
+    - name: "CN"
+    - uid: "DN"
+    - annotations:
+      - syncrecord: "syncRecord"
+```
+1. The filter that will retrieve all of the first-class groups that this sync will find relevant.
+* The the specific filter that will, when conjoined with the common filter, return only a specific group
+* The method for extracting data from an OpenShift `Group` to populate the specific filter
+* The directions for building the value of the specific filter from an LDAP group name
+* The the specific filter that will return only a specific member when the member annotation from a first-class group is injected and the resulting filter is conjoined with the common filter
+* The attribute on the first-class group entries in the LDAP server that holds their members
+* The specification for how to transfer attributes from a first-class group entity on the LDAP server to fields in an OpenShift `Group`
+
+##  Determining Which Groups To Sync
+An `LDAPGroupLister` determines what LDAP groups needs to be synced and outputs the result as a set of strings which encode the value of the unique filter needed to determine those groups from the LDAP server. For sync jobs where an LDAP request is necessary to determine which groups to sync, the `LDAPGroupLister` will populate the cache used by the `LDAPGroupDataExtractor` as the `LDAPDataSource`.
+
+```go
+// LDAPGroupLister lists the specific group filters for all groups that should be synced
+type LDAPGroupLister interface {
+  ListGroups() (groupFilters util.StringSet, err error)
+}
+```
+
+AN `LDAPGroupMapper` determines what OpenShift `Group` is being referred to by a filter designed for an LDAP group. This mapper finds the OpenShift resource to be updated or creates one if none exists yet.
+```go
+// LDAPGroupMapper maps a set of metainformation to an OpenShift Group
+type LDAPGroupMapper interface{
+  MapLDAPGroup(meta *kapi.ObjectMeta) (group *Group)
+}
+```
+
```

Comment:
I think I can live with an enumerated list of the different permutations of `LDAPGroupLister`:
 1. all applicable openshift groups to ldap groups
 1. some openshift groups to ldap groups
 1. all available ldap groups
 1. whitelist of ldap groups